doc/nrf/protocols/matter/getting_started/matter_api.rst

```diff
@@ -0,0 +1,248 @@
+
+.. code-block:: C++
+
+   using CustomInit = CHIP_ERROR (*)(void);
+
+   /** @brief Matter initialization data.
+    *
+    * This structure contains all user specific implementations of Matter interfaces
+    * and custom initialization callbacks that must be initialized in the Matter thread.
+    */
+   struct InitData {
+     /** @brief Matter stack events handler. */
+     chip::DeviceLayer::PlatformManager::EventHandlerFunct mEventHandler{ DefaultEventHandler };
+     /** @brief Pointer to the user provided NetworkCommissioning instance. */
+   #ifdef CONFIG_CHIP_WIFI
+     chip::app::Clusters::NetworkCommissioning::Instance *mNetworkingInstance{ &sWiFiCommissioningInstance };
+   #else
+     chip::app::Clusters::NetworkCommissioning::Instance *mNetworkingInstance{ nullptr };
+   #endif
+     /** @brief Pointer to the user provided custom server initialization parameters. */
+     chip::CommonCaseDeviceServerInitParams *mServerInitParams{ &sServerInitParamsDefault };
+     /** @brief Pointer to the user provided custom device info provider implementation. */
+     chip::DeviceLayer::DeviceInfoProviderImpl *mDeviceInfoProvider{ nullptr };
+   #ifdef CONFIG_CHIP_FACTORY_DATA
+     /** @brief Pointer to the user provided FactoryDataProvider implementation. */
+     chip::DeviceLayer::FactoryDataProviderBase *mFactoryDataProvider{ &sFactoryDataProviderDefault };
+   #endif
+   #ifdef CONFIG_CHIP_CRYPTO_PSA
+     /** @brief Pointer to the user provided OperationalKeystore implementation. */
+     chip::Crypto::OperationalKeystore *mOperationalKeyStore{ &sOperationalKeystoreDefault };
+   #endif
+     /** @brief Custom code to execute in the Matter main event loop before the server initialization. */
+     CustomInit mPreServerInitClbk{ nullptr };
+     /** @brief Custom code to execute in the Matter main event loop after the server initialization. */
+     CustomInit mPostServerInitClbk{ nullptr };
+
+     /** @brief Default implementation static objects that will be stripped by the compiler when above
+      * pointers are overwritten by the application. */
+   #ifdef CONFIG_CHIP_WIFI
+     static chip::app::Clusters::NetworkCommissioning::Instance sWiFiCommissioningInstance;
+   #endif
+     static chip::CommonCaseDeviceServerInitParams sServerInitParamsDefault;
+   #ifdef CONFIG_CHIP_FACTORY_DATA
+     static chip::DeviceLayer::FactoryDataProvider<chip::DeviceLayer::InternalFlashFactoryData> sFactoryDataProviderDefault;
+   #endif
+   #ifdef CONFIG_CHIP_CRYPTO_PSA
+     static chip::Crypto::PSAOperationalKeystore sOperationalKeystoreDefault;
+   #endif
+   };
+
+In this structure, each Matter component is defined as a public pointer to its generic interface that can be set with the concrete object.
+Moreover, each pointer is initially assigned with the default concrete implementation of the given interface that is stripped by the compiler in case the user overwrites it with customized implementation in the application.
```

Comment:
```suggestion
Each pointer is initially assigned with the default concrete implementation of the given interface.
The implementation is stripped by the compiler if the user overwrites it with a customized implementation in the application.
```
Is this still the correct meaning?

Reply:
Yes, I think it's fine. I will tune it a bit though.